packages/react-data-grid/src/HeaderRow.js

```diff
@@ -116,28 +109,46 @@ class HeaderRow extends React.Component {
   };
 
   getCells = (): Array<HeaderCell> => {
-    let cells = [];
-    let frozenCells = [];
-    for (let i = 0, len = columnUtils.getSize(this.props.columns); i < len; i++) {
-      let column = Object.assign({ rowType: this.props.rowType }, columnUtils.getColumn(this.props.columns, i));
-      let _renderer = this.getHeaderRenderer(column);
-      if (column.key === 'select-row' && this.props.rowType === 'filter') {
-        _renderer = <div></div>;
-      }
-      let HeaderCell = column.draggable ? this.props.draggableHeaderCell : BaseHeaderCell;
-      let cell = (
-        <HeaderCell
-          ref={(node) => this.cells[i] = node}
-          key={i}
-          height={this.props.height}
-          column={column}
-          renderer={_renderer}
-          resizing={this.props.resizing === column}
-          onResize={this.props.onColumnResize}
-          onResizeEnd={this.props.onColumnResizeEnd}
-          onHeaderDrop={this.props.onHeaderDrop}
+    const cells = [];
+    const frozenCells = [];
+    const { columns, rowType } = this.props;
+
+    for (let i = 0, len = columnUtils.getSize(columns); i < len; i++) {
+      const column = Object.assign({ rowType }, columnUtils.getColumn(columns, i));
```

Comment:
You've used the spread operator in `DraggableContainer.js`, why not here as well?

Reply:
I just kept the old code. I have replaced it with the spread operator